packages/medusa/src/api/routes/admin/products/update-product.ts

```diff
@@ -125,31 +133,74 @@ export default async (req, res) => {
 
   const manager: EntityManager = req.scope.resolve("manager")
   await manager.transaction(async (transactionManager) => {
+    const productServiceTx = productService.withTransaction(transactionManager)
+
     const { variants } = validated
     delete validated.variants
 
-    await productService
-      .withTransaction(transactionManager)
-      .update(id, validated)
+    const product = await productServiceTx.update(id, validated)
 
     if (!variants) {
       return
     }
 
-    const product = await productService
+    const variantRepo = manager.getCustomRepository(productVariantRepo)
+    const productVariants = await productVariantService
       .withTransaction(transactionManager)
-      .retrieve(id, {
-        relations: ["variants"],
-      })
+      .list(
```

Comment:
question: wouldn't this be paginated? Are we safeguading cases where product variants can be > 20?

Reply:
Maybe I'm missing something here. But doesn't this mean that its paginated? How are we skipping pagination here?

```
async list(
    selector: FilterableProductVariantProps,
    config: FindConfig<ProductVariant> & PriceSelectionContext = {
      relations: [],
      skip: 0,
      take: 20,
    }
  )
```

Reply:
When you pass a config object, the default value does not take place since there is a value that has been passed